experiment/metrics-server/main.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/metrics"
+)
+
+var (
+	prowJobs = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "prowjobs",
+		Help: "Number of prowjobs in the system",
+	}, []string{
+		// agent of the projob
+		"agent",
+		// state of the prowjob
+		"state",
```

Comment:
Should we parameterize this by `Spec.Job` and `Kind` as well? It would be useful to filter the metrics to see results for a specific job, jobs that match a regex, or jobs of a specific kind.

Reply:
I am going to add a `job` label. Not sure what you mean with `kind`.

Reply:
I'm assuming presubmit/postsubmit/periodic/batch ?

Reply:
SGTM - adding it as `type` though to match the prowjob field.